Microsoft.Toolkit.Uwp.UI.Controls/MasterDetail/MasterDetail.cs

```diff
@@ -0,0 +1,383 @@
+        }
+
+        /// <summary>
+        ///     Gets or sets the detail template.
+        /// </summary>
+        /// <value>
+        ///     The detail template.
+        /// </value>
+        public DataTemplate DetailTemplate
+        {
+            get { return (DataTemplate)GetValue(DetailTemplateProperty); }
+            set { SetValue(DetailTemplateProperty, value); }
+        }
+
+        /// <summary>
+        ///     Gets or sets the detail template selector.
+        /// </summary>
+        /// <value>
+        ///     The detail template selector.
+        /// </value>
+        public DataTemplateSelector DetailTemplateSelector
+        {
+            get { return (DataTemplateSelector)GetValue(DetailTemplateSelectorProperty); }
+            set { SetValue(DetailTemplateSelectorProperty, value); }
+        }
+
+        /// <summary>
+        ///     Gets or sets the display mode.
+        /// </summary>
+        /// <value>
+        ///     The display mode.
+        /// </value>
+        public MasterDetailDisplayMode DisplayMode
+        {
+            get { return (MasterDetailDisplayMode)GetValue(DisplayModeProperty); }
+            set { SetValue(DisplayModeProperty, value); }
+        }
+
+        /// <summary>
+        ///     Gets or sets the display visible.
+        /// </summary>
+        /// <value>
+        ///     The display visible.
+        /// </value>
+        public MasterDetailDisplayVisible DisplayVisible
+        {
+            get { return (MasterDetailDisplayVisible)GetValue(DisplayVisibleProperty); }
+            set { SetValue(DisplayVisibleProperty, value); }
+        }
+
+        private static void OnDisplayVisibleChanged(DependencyObject sender, DependencyPropertyChangedEventArgs e)
+        {
+            if ((MasterDetailDisplayVisible)e.OldValue != (MasterDetailDisplayVisible)e.NewValue)
+            {
+                (sender as MasterDetail)?.SendDisplayVisibleChangedEvent();
+            }
+        }
+
+        /// <summary>
+        ///     Occurs when [display visible changed].
+        /// </summary>
+        public event EventHandler<DisplayVisibleArgs> DisplayVisibleChanged;
+
+        /// <summary>
+        ///     Invoked whenever application code or internal processes (such as a rebuilding layout pass) call ApplyTemplate. In
+        ///     simplest terms, this means the method is called just before a UI element displays in your app. Override this method
+        ///     to influence the default post-template logic of a class.
+        /// </summary>
+        protected override void OnApplyTemplate()
+        {
+            base.OnApplyTemplate();
+
+            Update();
+        }
+
+        private void OnUnloaded(object sender, RoutedEventArgs routedEventArgs)
+        {
+            if (ApiInformation.IsTypePresent("Windows.Phone.UI.Input.HardwareButtons"))
+            {
+                HardwareButtons.BackPressed -= HardwareButtons_BackPressed;
+            }
+
```

Comment:
Attached sample.. tested on Desktop, emulator and 930 running prod OS release
[BackRequestedSample.zip](https://github.com/Microsoft/UWPCommunityToolkit/files/533720/BackRequestedSample.zip)